scipy/stats/_hypotests.py

```diff
@@ -130,3 +131,186 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
+    Let `count1` and `count1` be the observed values of :math:`X_1` and :math:`X_2`,
+    respectively. The problem of interest here is to test
+    
+    .. math::
+       H_0: \lambda_1 - \lambda_2 \le \mathtt{diff} \quad vs. \quad
+       H_a: \lambda_1 - \lambda_2 > \mathtt{diff}
+    
+    for right sided `greater` hypothesis where :math:`\mathtt{diff} \ge 0` is a given
+    number, based on (`nobs1`, `count1`, `nobs2`, `count2`). `two-sided` and `greater`
+    cases are demonstrated by [1]_.The `less` hypothesis performed by switching the
+    arguments on `greater` hypothesis.
+
+    Parameters
+    ----------
+    count1, count2 : int
+        Count event of interest from the first and second samples respectively
+    nobs1, nobs2 : int
+        Sample size observed
+    diff : int of float, optional
+        The difference of mean between two samples under null hypothesis
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Which alternative hypothesis to test
+
+    Returns
+    -------
+    statistic : float
+        The test statistic calculated from observed samples
+    pvalue : float
+        The associated p-value based on estimated p-value of the standardized
+        difference.
+
+    Notes
+    -----
+    The Poisson distribution is commonly used to model many processes such as
+    transactions per user. A simple test to compare difference between two
+    means of Poisson samples is C-test, based on conditional distribution.
+    Meanwhile the E-test is an unconditional test
+
+    Based the author results [1]_, E-test is more powerful than C-test. The E-test
+    is almost exact because the test exceed the nominal value only by negligible
+    amount. Compared to C-test which produce smaller size than nominal value.
+
+    References
+    ----------
+    .. [1]  Krishnamoorthy, K., & Thomson, J. (2004). A more powerful test for
+       comparing two Poisson means. Journal of Statistical Planning and Inference,
+       119(1), 23-35.
+    .. [2]  Przyborowski, J., & Wilenski, H. (1940). Homogeneity of results in
+       testing samples from Poisson series: With an application to testing clover
+       seed for dodder. Biometrika, 31(3/4), 313-323.
+
+    Examples
+    --------
+    >>> from scipy import stats
+
+    Taken from Przyborowski and Wilenski (1940) [2]_. Suppose that a purchaser wishes to
+    test the number of dodder seeds (a weed) in a sack of clover seeds that he bought
+    from a seed manufacturing company. A 100 g sample is drawn from a sack of clover
+    seeds prior to being shipped to the purchaser. The sample is analyzed and found to
+    contain no dodder seeds; that is, k1 = 0. Upon arrival, the purchaser also draws
+    a 100 g sample from the sack. This time, three dodder seeds are found in the sample;
+    that is, k2 = 3. The purchaser wishes to determine if the difference between the
+    samples could not be due to chance.
+
+    >>> stats.poisson_means_test(0, 100, 3, 100)
```

Comment:
maybe use 
>> w, p = stats.poisson_means_test,
>> w, p
-1.7320508075688772, 0.08837900929018155

otherwise the output is a bit hard to read